### ISSUE TITLE
Unhelpful timeout exception

### ISSUE THREAD

**Issue by tomasherman** [http4s/http4s#3087]
Hello, i found another fairly unhelpful exception coming from http4s:

```
java.util.concurrent.TimeoutException: Request timeout after 10000 ms
	at org.http4s.client.blaze.BlazeClient$$anon$1.run(BlazeClient.scala:152)
	at monix.execution.schedulers.TracingRunnable.run(TracingRunnable.scala:33)
	at monix.execution.internal.InterceptRunnable.run(InterceptRunnable.scala:27)
	at monix.execution.internal.forkJoin.AdaptedForkJoinTask.exec(AdaptedForkJoinTask.scala:27)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

It would be great if we could add some context - like request that failed and maybe even IP address of the upstream that timeout out.

**Comment by tomasherman** [comment]
I was hoping to help out by providing a PR but i have a hard time to find all the pieces ... i think what needs to be done is to add information about IP address to `BlazeConnection`? 

Seems like this would be a good place to put that information in: https://github.com/http4s/http4s/blob/master/blaze-client/src/main/scala/org/http4s/client/blaze/Http1Support.scala#L64 ?

**Comment by tomasherman** [comment]
So there is a bunch of exceptions in `BlazeClient` ... i wonder if it would make sense to add to all of them at least the `RequestKey` ... thoughts?

**Comment by rossabaker** [comment]
Yes, you could enhance the exceptions by pattern matching [Lefts]( https://github.com/http4s/http4s/blob/efa36675a8dd06aee80f5bb908899cbcbc8ce442/blaze-client/src/main/scala/org/http4s/client/blaze/Http1Support.scala#L72).  It will be the same basic tradeoffs as on the other: using the standard Java exception vs. a richer wrapper, with some opportunities to extend the standard Java exceptions like we did in 0.20.

**Comment by rossabaker** [comment]
Looked a bit closer.  I think the timeout you're looking for is [here](https://github.com/http4s/http4s/blob/a97a153931e324a80a27e4b99129042272c81eac/blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala#L152).  The `RequestKey` should be in scope, and we can extend that message.

**Comment by tomasherman** [comment]
Sure. The one thing I would like to achieve is to have inetaddres available
for timeouts as well. But it's going to be more difficult I think. Is it
possible to get upstream address from connection? Is it something you would
be ok with adding?

**Comment by tomasherman** [comment]
I decided to improve (imho) some logging that we encountered in the wild and would benefited from, hope it's ok

**Comment by tomasherman** [comment]
fixed in #3089
